dev/NavigationView/NavigationView.cpp

```diff
@@ -3884,34 +4428,111 @@ winrt::IInspectable NavigationView::GetItemFromIndex(const winrt::ItemsRepeater&
     return nullptr;
 }
 
-winrt::NavigationViewItemBase NavigationView::GetContainerForIndexPath(const winrt::IndexPath& indexPath)
+winrt::IndexPath NavigationView::GetIndexPathOfItem(const winrt::IInspectable& data)
 {
-    if (indexPath && indexPath.GetSize() > 0)
+    if (auto const nvib = data.try_as<winrt::NavigationViewItemBase>())
     {
-        auto index = indexPath.GetAt(0);
-        if (IsTopNavigationView())
-        {
-            // Get the repeater that is presenting this item
-            auto ir = m_topDataProvider.IsItemInPrimaryList(index) ? m_topNavRepeater.get() : m_topNavRepeaterOverflowView.get();
-
-            // Get the index of the item in the repeater
-            auto irIndex = m_topDataProvider.ConvertOriginalIndexToIndex(index);
+        return GetIndexPathForContainer(nvib);
+    }
 
-            // Get the container of the item
-            if (auto container = ir.TryGetElement(irIndex))
+    // In the databinding scenario, we need to conduct a search where we go through every item,
+    // realizing it if necessary.
+    if (IsTopNavigationView())
+    {
+        // TODO
```

Comment:
>// TODO [](start = 8, length = 7)

Is this okay to be left as a TODO?

Reply:
@ojhad Please create an issue and link it here.

Reply:
Pushing the commit for this fix. It was just existing functionality that had not been hooked up here.